pkg/testing/matrix/matrix.go

```diff
@@ -0,0 +1,457 @@
+		case GO:
+			projectGenerator = gogen.GenerateProject
+		case NODEJS:
+			projectGenerator = jsgen.GenerateProject
+		case PYTHON:
+			projectGenerator = pygen.GenerateProject
+		case JAVA:
+			projectGenerator = func(directory string, project workspace.Project,
+				p *pcl.Program, localProjects map[string]string) error {
+				return javagen.GenerateProject(directory, project, p)
+			}
+		case YAML: // nolint: goconst
+			projectGenerator = func(directory string, project workspace.Project,
+				p *pcl.Program, localProjects map[string]string) error {
+				return yamlgen.GenerateProject(directory, project, p)
+			}
+		default:
+			assert.FailNow(t, "Unsupported language: "+langOpt.Language)
+		}
+
+		subdir := filepath.Join(dir, langOpt.Language)
+		//check if subdir exists
+		if _, err := os.Stat(subdir); !os.IsNotExist(err) {
+			assert.NoError(t, os.RemoveAll(subdir))
+		}
+
+		//create subdir
+		err = os.MkdirAll(subdir, 0755)
+		assert.NoError(t, err)
+
+		//generate project
+		err = projectGenerator(subdir, *proj, pclProgram, localProjects)
+		assert.NoError(t, err)
+
+		//Configure opts
+		langPtOpts := i.ProgramTestOptions{}
+		if opts.Program != nil {
+			langPtOpts = *opts.Program
+		}
+		if langOpt.Opts != nil {
+			langPtOpts = langPtOpts.With(*langOpt.Opts)
+		}
+		langPtOpts = langPtOpts.With(i.ProgramTestOptions{
+			Dir: subdir,
+		})
+		t.Run(langOpt.Language, func(t *testing.T) {
+			i.ProgramTest(t, &langPtOpts)
+		})
+		//clean up subdir
+
+		//Including this seems to cause all tests to fail.
+		/*t.Cleanup(func() {
+			os.RemoveAll(subdir)
+		})*/
+	}
+}
+
+func PythonConfigurePkg(pkg *schema.Package) error {
+	pkg.Version = &semver.Version{
+		Major: pkg.Version.Major,
+		Minor: pkg.Version.Minor,
+		Patch: pkg.Version.Patch,
+	} // Prune patch and build versions, since semver and python don't seem to agree on formatting.
+
+	raw, notnil := pkg.Language["python"]
+	message, ismessage := raw.(json.RawMessage)
+	var pyPkg pygen.PackageInfo
+	if notnil && ismessage {
+		err := json.Unmarshal(message, &pyPkg)
+		if err != nil {
+			return err
+		}
+	} else if notnil && !ismessage {
+		info, ok := raw.(pygen.PackageInfo)
+		if !ok {
+			return fmt.Errorf("invalid python package info")
+		}
+		pyPkg = info
+	} else if !notnil {
+		pyPkg = pygen.PackageInfo{}
+	}
+	pyPkg.RespectSchemaVersion = true
+	pkg.Language["python"] = pyPkg
+	return nil
+}
+
+func NodeConfigurePkg(pkg *schema.Package) error {
+	raw, notnil := pkg.Language["nodejs"]
+	message, ismessage := raw.(json.RawMessage)
+	var nodePkg jsgen.NodePackageInfo
+	if notnil && ismessage {
+		err := json.Unmarshal(message, &nodePkg)
+		if err != nil {
+			return err
+		}
+	} else if notnil && !ismessage {
+		info, ok := raw.(jsgen.NodePackageInfo)
+		if !ok {
+			return fmt.Errorf("invalid nodejs package info")
+		}
+		nodePkg = info
+	} else if !notnil {
+		nodePkg = jsgen.NodePackageInfo{}
+	}
+	nodePkg.RespectSchemaVersion = true
```

Comment:
Why do node and python need this but nothing else? That seems inconsistent.

Reply:
Otherwise it just says ${VERSION} and pip/node freak out